lib/housekeeping.js

```diff
@@ -58,53 +60,66 @@ class HouseKeeper {
     //       }
     //     },
     //     ...
-    //   ]
+    //   ],
+    //   ${region} : [
+    //   ...
+    //   ],
+    //   ...
     // }
 
     //assert(typeof totalsByCategory === 'object');
 
-    //assert(Object.keys(totalsByCategory).length === 1);
-    let region =  Object.keys(totalsByCategory)[0];
+    // Keep track of overall totals, over all regions
+    let totalGbUsage = 0;
+    let totalCount = 0;
+
+    Object.keys(totalsByCategory).forEach(function(region) {
     
-    let volumeTypes = totalsByCategory[region];
-    //assert(typeof volumeTypes === 'object');
-    //assert(typeof volumeTypes.hasOwnProperty('length'));
+      let volumeTypes = totalsByCategory[region];
+      //assert(typeof volumeTypes === 'object');
+      //assert(typeof volumeTypes.hasOwnProperty('length'));
 
-    volumeTypes.forEach(function(data) {
-      //assert(typeof data === 'object');
+      volumeTypes.forEach(function(data) {
+        //assert(typeof data === 'object');
       
-      //assert(Object.keys(data).length === 1);
-      let type = Object.keys(data)[0];
-
-      //assert(data.hasOwnProperty('active'));
-      //assert(data.hasOwnProperty('unused'));
-      let {active, unused} = data;
-
-      //assert(typeof active === 'object');
-      //assert(active.hasOwnProperty('gb'));
-      //assert(active.hasOwnProperty('count'));
-
-      this.monitor.count('ebs-volumes:${region}:gb-usage', active.gb);
-      this.monitor.count('ebs-volumes:${region}:count', active.count);
-
-      //assert(typeof unused === 'object');
-      //assert(unused.hasOwnProperty('gb'));
-      //assert(unused.hasOwnProperty('count'));
-
-      this.monitor.count('ebs-volumes:${region}:unused-gb-usage', unused.gb);
-      this.monitor.count('ebs-volumes:${region}:unused-count', unused.count);
-
-      let totalGbUsage = active.gb + unused.gb;
-      let totalCount = active.count + unused.count;
-      this.monitor.count('ebs-volumes:global:gb-usage', totalGbUsage);
-      this.monitor.count('ebs-volumes:global:count', totalCount);
-      this.monitor.count('ebs-volumes:${region}:type:${type}:gb-usage', totalGbUsage);
-      this.monitor.count('ebs-volumes:${region}:type:${type}:count', totalCount);
+        //assert(Object.keys(data).length === 1);
+        let type = Object.keys(data)[0];
+
+        //assert(data.hasOwnProperty('active'));
+        //assert(data.hasOwnProperty('unused'));
+        let {active, unused} = data;
+
+        //assert(typeof active === 'object');
+        //assert(active.hasOwnProperty('gb'));
+        //assert(active.hasOwnProperty('count'));
+
+        this.monitor.count('ebs-volumes:${region}:gb-usage', active.gb);
```

Comment:
`this` appears to be undefined in `_submitVolumeTotals()`. When I get rid of `this` in `this.monitor.count()`, my tests in https://github.com/rstiyer/ec2-manager/pull/3 are able to run. I'm guessing it's because `_submitVolumeTotals()` is called by `_handleVolumeTotals()` as `this._submitVolumeTotals()`.

The `this` objects in JS always get me confused so that may or may not be the right solution.

Reply:
It's interesting that getting rid of `this` makes it work, since we call all functions with `this` (e.g. `sweep` calls `this._handleVolumeData` which in turn calls `this._submitVolumeTotals`). I guess I'll need to look into this.

Reply:
It's also weird that we don't need to get rid of the `this` on lines 116 and 117. Maybe it has something more to do with the nested loops.

Reply:
I changed the function in `foreach()` to an arrow function (as per @djmitche's suggestions), and the `this` object is now defined! I also changed it in `_calculateVolumeTotals()` in this commit: https://github.com/SamanthaYu/ec2-manager/commit/5df323c7181b1544d3bfad348bd5c35d105009f9